File: Iterator/Python/ListaPalabrasIterador.py
from decimal import Clamped
from Iterador import *

class ListaPalabrasIterador(Iterador):
    
    def __init__(self, p1, p2, p3, p4, p5):
        self._palabra1 = p1
        self._palabra2 = p2
        self._palabra3 = p3 
        self._palabra4 = p4 
        self._palabra5 = p5
        self._posicion = 0
    
    def siguiente(self):
        p = ''
        if self._posicion == 0:
            p = self._palabra1
        if self._posicion == 1:
            p =  self._palabra2        
        if self._posicion == 2:
            p =  self._palabra3
        if self._posicion == 3:
            p = self._palabra4
        if self._posicion == 4:
            p =  self._palabra5

        self._posicion += 1
        return p
    def tieneSiguiente(self):
        
        if self._posicion < 5:
            return True
        else:
            return False

File: Iterator/Java/ListaPalabrasIterador.java
public class ListaPalabrasIterador implements Iterador
{
    private String palabra1, palabra2, palabra3, palabra4, palabra5;
    private int posicion;

    ListaPalabrasIterador(String p1, String p2, String p3, String p4, String p5)
    {
        this.palabra1 = p1;
        this.palabra2 = p2;
        this.palabra3 = p3;
        this.palabra4 = p4;
        this.palabra5 = p5;
        this.posicion = 0;
    }
    @Override
    public Object siguiente() {
        switch(this.posicion++)
        {
            case 0:
                return this.palabra1;
                
            case 1:
                return this.palabra2;
                
            case 2:
                return this.palabra3;
                
            case 3:
                return this.palabra4;
                
            case 4:
                return this.palabra5;
        }
        return null;
    }

    @Override
    public boolean tieneSiguiente() {
        if(this.posicion < 5)
        {
            return true;
        }else{
            return false;
        }
    }
    
}
